src/node/nodestate.h

```diff
@@ -326,13 +326,22 @@ namespace ccf
 
       // Peer certificate needs to be signed by network certificate
       auto tls_ca = std::make_shared<tls::CA>(args.network_cert);
-      auto join_client_cert = std::make_shared<tls::Cert>(
+      auto join_client_cert = std::make_unique<tls::Cert>(
         Actors::NODES, tls_ca, node_cert, node_kp.private_key(), nullb);
 
       // Create and connect to endpoint
-      auto join_client = rpcsessions.create_client(join_client_cert);
+      auto join_client =
+        rpcsessions.create_client(rpc_ctx, std::move(join_client_cert));
+
+      // If the join network command is synchronous, only reply to the client
+      // when the leader has responded
+      if (args.is_sync)
+        rpc_ctx.is_suspended = true;
```

Comment:
This flag used to be called `is_forwarded` but needed renaming as the feature is now more generic. Any better names for this flag, please let me know.